ops/model.py

```diff
@@ -1771,6 +1782,71 @@ def remote_model(self) -> RemoteModel:
             self._remote_model = RemoteModel(uuid=d['uuid'])
         return self._remote_model
 
+    def load_data(
+        self,
+        cls: type[_InterfaceType],
+        app_or_unit: Union[Unit, Application],
+        *args: Any,
+        decoder: Callable[[Any], Any] | None = None,
+        encoder: Callable[[Any], Any] | None = None,
```

Comment:
Comment from spec review: these could be tighter. The return type will be `str` because that's all that Juju will accept. The argument probably has to stay as `Any`, I think.